plugins/env/app/controllers/environment_variable_groups_controller.rb

```diff
@@ -4,7 +4,18 @@ class EnvironmentVariableGroupsController < ApplicationController
   before_action :group, only: [:show]
 
   def index
-    @groups = EnvironmentVariableGroup.all
+    @groups = EnvironmentVariableGroup.all.includes(:environment_variables)
+    respond_to do |format|
+      format.html
+      format.json do
+        groups = @groups.map do |group|
+          json = group.as_json
+          json['variables'] = group.environment_variables.sort_by(&:id).map(&:name).uniq
```

Comment:
- prefer `variables` sounds like objects, maybe use `variable_names` ?
 - copy-pasted from the view ... turn into a method on the model and reuse ?
 - prefer to add to as_json and then use `render_as_json :environment_variable_groups, @groups` so we get consistently formatted replies

Reply:
```Ruby
def variable_names
  ...
end

def as_json
  super(methods: [:variable_names])
end
```